test/functional/wallet_upgradewallet.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+# Copyright (c) 2018-2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""upgradewallet RPC functional test
+
+Test upgradewallet RPC. Download v0.16.3 node binary:
+
+contrib/devtools/previous_release.sh -b v0.16.3
+"""
+
+import os
+import shutil
+
+from test_framework.test_framework import BitcoinTestFramework, SkipTest
+from test_framework.util import (
+    adjust_bitcoin_conf_for_pre_16_3,
+    assert_equal,
+    assert_greater_than
+)
+
+class UpgradeWalletTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 2
+        self.extra_args = [
+            ["-addresstype=bech32"],
+            []
+        ]
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def setup_nodes(self):
+        if os.getenv("TEST_PREVIOUS_RELEASES") == "false":
+            raise SkipTest("upgradewallet RPC tests")
+
+        releases_path = os.getenv("PREVIOUS_RELEASES_DIR") or os.getcwd() + "/releases"
+        if not os.path.isdir(releases_path):
+            if os.getenv("TEST_PREVIOUS_RELEASES") == "true":
+                raise AssertionError("TEST_PREVIOUS_RELEASES=1 but releases missing: " + releases_path)
+            raise SkipTest("This test requires binaries for previous releases")
+
+        self.add_nodes(self.num_nodes, extra_args=self.extra_args, versions=[
+            None,
+            160300
```

Comment:
would it be possible to user version 15, so that `-usehd=0` can be passed to the wallet?

Reply:
Have added a v0.15.2 node, so that we are now testing v0.15.2 with *-usehd=0* and v0.16.3 with *-usehd=1*.

Reply:
Btw. there is now a new helper method in *util.py* called `get_nodes_least_version`, because the framework couldn't start the v0.15.2 node. The error was thrown in `sync_mempools`and `sync_blocks`. It seems to have something to do with the call of `syncwithvalidationinterfacequeue`. This indirectly affected the `getpeerinfo`from `sync_blocks`, but because 0.15.2 never needed any peers for these particular tests, I thought, maybe a small helper methode for filtering those old nodes would be sufficient. If not, please, let me know what alternatives we have here.